### PR TITLE
Implement R8 and RG8 texture formats with uint and unorm components.

### PR DESCRIPTION
**Pull request by SenorBlanco**
Backend support added for Metal, D3D12 and OpenGL.
Also fix BGRA support in OpenGL to give it a valid internal format.

### CODE REVIEW

File: src/backend/opengl/TextureGL.cpp [hunk 1]
```diff
@@ -38,9 +38,11 @@ namespace backend { namespace opengl {
                     return {GL_RGBA8, GL_RGBA, GL_UNSIGNED_BYTE};
                 case nxt::TextureFormat::R8G8B8A8Uint:
                     return {GL_RGBA8UI, GL_RGBA, GL_UNSIGNED_INT};
+                case nxt::TextureFormat::R8Unorm:
+                    return {GL_R8, GL_RED, GL_UNSIGNED_BYTE};
                 case nxt::TextureFormat::B8G8R8A8Unorm:
-                    // This doesn't have an enum for the internal format in OpenGL.
-                    return {GL_NONE, GL_BGRA, GL_UNSIGNED_BYTE};
+                    // This doesn't have an enum for the internal format in OpenGL, so use RGBA8.
```
Comment (Kangz) [inline src/backend/opengl/TextureGL.cpp]: Can we have this fix in a separate commit?

**Comment by Kangz** [comment]
LGTM too

File: next.json [hunk 2]
```diff
@@ -1200,7 +1200,8 @@
             {"value": 0, "name": "r8 g8 b8 a8 unorm"},
             {"value": 1, "name": "r8 g8 b8 a8 uint"},
             {"value": 2, "name": "b8 g8 r8 a8 unorm"},
-            {"value": 3, "name": "d32 float s8 uint"}
+            {"value": 3, "name": "d32 float s8 uint"},
+            {"value": 4, "name": "r8 unorm"}
```
Comment (Kangz) [inline next.json]: Maybe this could complete the [unorm, uint] x [1, 2, 4 component] matrix.

File: src/backend/opengl/TextureGL.cpp (hunk 1 above)
Comment (SenorBlanco) [reply src/backend/opengl/TextureGL.cpp]: Done.

File: next.json (hunk 2 above)
Comment (SenorBlanco) [reply next.json]: Done.

**Comment by Kangz** [comment]
Code simplification and update LGTM, can you squash this in a single commit?

**Comment by kainino0x** [comment]
Can't we just "Squash and merge" on GitHub?

**Comment by Kangz** [comment]
Squashing causes the commit message of all temporary commits to be included which is a bit ugly.

**Comment by kainino0x** [comment]
Squash and merge button lets you rewrite the commit message (though it still means the merger, rather than the committer, has to do it)

**Comment by SenorBlanco** [comment]
Squashed.

**Pull request merged**